academy/3-ibc/4-light-client-dev.md

```diff
@@ -0,0 +1,177 @@
+
+Prior to the client refactor (prior to v7) the client and consensus states are set within the `02-client` submodule. Moving these responsibilities from the `02-client` to the individual light client implementations, including the setting of updated client state and consensus state in the client store, provides light client developers with a greater degree of flexibility with respect to storage and verification.
+
+</HighlightBox>
+
+### Packet commitment verification
+
+Next to updating the client and consensus state, the light client also provides functionality to perform the verification required for the packet flow (send, receive and acknowledge or timeout). IBC uses Merkle proofs to verify against the trusted root if state is either committed or absent on a predefined standardized key path, as defined in [ICS-24 host requirements](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements).
+
+As you have seen in the [previous section on clients](./4-clients.md), when the IBC handler receives a message to receive, acknowledge or timeout a packet, it will call one of the following functions on the `connectionKeeper` to verify if the remote chain includes (or does not include) the appropriate state:
+
+* [`VerifyPacketCommitment`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/03-connection/keeper/verify.go#L205) checks if the proof added to a `MsgRecvPacket` submitted to the destination, points to a valid packet commitment on the source.
+* [`VerifyPacketAcknowledgement`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/03-connection/keeper/verify.go#L250) checks if the proof added to a `MsgAcknowledgePacket` submitted to the source, points to a valid packet receipt commitment on the destination.
+* [`VerifyPacketReceiptAbsence`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/03-connection/keeper/verify.go#L296) checks if the proof added to a `MsgTimeout` submitted to the source, proves that a packet receipt is absent at a height beyond the timeout height on the destination.
+
+All of the above rely on either `VerifyMembership` or `VerifyNonMembership` methods to prove either inclusion (also referred to as _existence_) or non-inclusion (_non-existence_) at a given commitment path.
+
+It is up to the light client developer to add these methods to their light client implementation.
+
+<HighlightBox type="reading">
+
+Please refer to the [ICS-23 implementation](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/23-commitment/types/merkle.go#L131-L205) for a concrete example.
+
+</HighlightBox>
+
+### Add light client to Cosmos SDK chain
+
+Now suppose that you have managed to develop a light client implementation fulfiling the requirements described above. How do you now get the light client on chain?
+
+This will depend on the chain environment you are in, so from here on out the assumption will be made that you will want to add the light client module to a Cosmos SDK chain.
```

Comment:
```suggestion
This will depend on the chain environment you are in, so from here on out the assumption will be made that you will want to add the light client module to a Cosmos SDK chain, in order for the Cosmos SDK chain that you are interacting with to be able to verify proofs of the consensus state coming from your non-`07-tendermint` chain.

For example, if you are developing a light client which enables proof verification of ETH2 or Solana, you would need to deploy a light client that is able to verify proofs of ETH2 or Solana consensus state on Cosmos SDK chains which you want to interoperate with.
```